compiler/z/codegen/OMRMemoryReference.cpp

```diff
@@ -3474,12 +3473,14 @@ reuseS390MemoryReference(TR::MemoryReference *baseMR, int32_t offset, TR::Node *
 TR::MemoryReference *
 generateS390MemoryReference(TR::Node * node, TR::SymbolReference * sr, TR::CodeGenerator * cg)
    {
+   TR_ASSERT_FATAL(node->getOpCode().hasSymbolReference(), "Memory reference generation API needs a node with symbol reference\n");
```

Comment:
This API has a SymbolReference passed in and does not use the `node`'s symbolRef. Hence, this is not a memory reference without symbol reference. Hence, this line shouldn't be added.